### DISCUSSION TITLE
Children components don't see update of one atom in parent one

### DISCUSSION THREAD

**Discussion by AlekseyP18** [pmndrs/jotai#2550] [Q&A]
## Summary
Hello dear jotai team. I liked this store manager because it has all needed features. I've been trying to migrate from mobx store to jotai, and I'm facing with one bug.

I have 2 components App and ParentFruit which use one atom - pricesAtom

When I added new price in App component the child component - ParentFruit doesn't see this new one object and doesn't render this new one

There is the same problem with arrays too - todosAtom

## Link to reproduction
[Link to sandbox reprodaction](https://codesandbox.io/p/sandbox/react-typescript-forked-gxtc9y?file=%2Fsrc%2FApp.tsx%3A8%2C33&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clw0zrsgw00063b6kg47v4d3w%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clw0zrsgw00023b6kp0ksodfu%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clw0zrsgw00033b6kyz4rgy4m%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clw0zrsgw00053b6k8ztmih02%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B56.39492326104097%252C43.60507673895903%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clw0zrsgw00023b6kp0ksodfu%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clw0zrsgw00013b6ktuo9yy22%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clw11ubkn00023b6kz1hn1w0h%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A8%252C%2522startColumn%2522%253A33%252C%2522endLineNumber%2522%253A8%252C%2522endColumn%2522%253A33%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clw0zrsgw00023b6kp0ksodfu%2522%252C%2522activeTabId%2522%253A%2522clw11ubkn00023b6kz1hn1w0h%2522%257D%252C%2522clw0zrsgw00053b6k8ztmih02%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clw0zrsgw00043b6ktf3j3afj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clw0zrsgw00053b6k8ztmih02%2522%252C%2522activeTabId%2522%253A%2522clw0zrsgw00043b6ktf3j3afj%2522%257D%252C%2522clw0zrsgw00033b6kyz4rgy4m%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clw0zrsgw00033b6kyz4rgy4m%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

## Check List

Please do not ask questions in issues.

- [ ] I've already opened a [discussion](https://github.com/pmndrs/jotai/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [ ] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

**Comment by rothsandro** [comment]
The problem is where your `Provider` is placed. The `useAtom` call in your `App` component is **not** wrapped in a provider and therefore uses the default store of Jotai. Your `ParentFruit` component is wrapped in a `Provider` and therefore uses a separate store, not the default store.

```tsx
const App = () => {
  const [...] = useAtom(fruitsAtom);    // (1) Default store

  return (
    <Provider>
      <ParentFruit />                   // (2) Store from Provider
      <button>Add Fruit</button>
    </Provider>
  );
};
```

The `Provider` should be at the root of your application, everything should be wrapped in the provider.


```tsx
const App = () => {
  return (
    <Provider>
      <DeliciousFruits />
    </Provider>
  );
}

const DeliciousFruits = () => {
  const [...] = useAtom(fruitsAtom);

  return (
    <ParentFruit />
    <button>Add Fruit</button>
  );
};
```

PS. I think it's better to open a [discussion](https://github.com/pmndrs/jotai/discussions) instead of an issue for the next time.
